app/controllers/char_attributes_controller.rb

```diff
@@ -1,8 +1,9 @@
 # Char_Attributes Controller
 class CharAttributesController < ApplicationController
+  load_and_authorize_resource :character
```

Comment:
@zzokki81 as the name says `load_and_authorize_resource` will load a resource. This means you don't have to manually load it like you are doing now at line 9 and other lines. Please go through all controllers and see where you can eliminate some of the manual loading (i.e. read up on `load_and_authorize_resource`).

Reply:
@zzokki81 I still wonder why you need set_character methods and set_char_attribute. Did you look at https://github.com/CanCanCommunity/cancancan#32-loaders and the rest of the documentation? Cancancan is supposed to load these for you, no?